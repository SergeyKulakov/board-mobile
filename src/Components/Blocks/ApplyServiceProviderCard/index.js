import Component from './ApplyServiceProviderCard'
import withNamespaces from 'Components/HOC/withNamespaces'

export default withNamespaces(Component)
