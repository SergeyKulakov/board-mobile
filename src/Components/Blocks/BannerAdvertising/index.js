import withNamespaces from 'Components/HOC/withNamespaces'

import BannerAdvertising from './BannerAdvertising'

export default withNamespaces(BannerAdvertising)
