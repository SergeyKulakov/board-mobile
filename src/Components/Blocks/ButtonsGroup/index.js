import withNamespaces from 'Components/HOC/withNamespaces'

import ButtonsGroup from './ButtonsGroup'

export default withNamespaces(ButtonsGroup)
