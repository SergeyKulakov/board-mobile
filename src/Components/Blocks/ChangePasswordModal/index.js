import { compose } from 'redux'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import withNamespaces from 'Components/HOC/withNamespaces'
import withPuck from 'Components/HOC/withPuck'

import { changePassword } from 'Redux/actions/settings'
import { getUser } from 'Redux/selectors/user'

import Component from './ChangePasswordModal'

const actions = { onSubmit: changePassword }

const selectors = createStructuredSelector({
  user: getUser,
})

export default compose(
  withNamespaces,
  withPuck,
  connect(
    selectors,
    actions,
  ),
)(Component)
