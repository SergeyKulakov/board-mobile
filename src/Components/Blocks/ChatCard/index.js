import React from 'react'
import PropTypes from 'prop-types'
import _ from 'lodash'
import { getShortUserName } from 'Helpers/user'
import { getMessageTime } from 'Helpers/chats'
import { ActivityIndicator } from 'react-native'
import { blockChatConst, unblockChatConst } from 'Constants/chats'

import {
  Container,
  AvatarWrapper,
  Content,
  TopPart,
  UserName,
  LastDate,
  LastMessage,
  NameWrapper,
  Avatar,
} from './style'

const ChatCard = ({ messages, partner, isChatBlocked, isLoading }) => (
  <Container>
    <AvatarWrapper>
      <Avatar avatarURL={partner.avatarURL} username={partner.username} />
    </AvatarWrapper>
    <Content>
      <TopPart>
        <NameWrapper>
          <UserName isChatBlocked={isChatBlocked}>
            {getShortUserName(
              partner.given_name,
              partner.family_name,
              partner.username,
            )}
          </UserName>
          {isLoading ? <ActivityIndicator /> : null}
        </NameWrapper>
        {_.isEmpty(messages) ? null : (
          <LastDate isChatBlocked={isChatBlocked}>
            {getMessageTime(messages[messages.length - 1].createdAt)}
          </LastDate>
        )}
      </TopPart>
      {_.isEmpty(messages)
        ? null
        : messages[messages.length - 1].type !== 'voice_message' && (
            <LastMessage
              isChatBlocked={isChatBlocked}
              isVisible={
                !(
                  messages[messages.length - 1].text === blockChatConst ||
                  messages[messages.length - 1].text === unblockChatConst
                )
              }
            >
              {messages[messages.length - 1].text}
            </LastMessage>
          )}
    </Content>
  </Container>
)

ChatCard.propTypes = {
  messages: PropTypes.array,
  partner: PropTypes.object,
  isChatBlocked: PropTypes.bool,
  isLoading: PropTypes.bool,
}

export { ChatCard }
