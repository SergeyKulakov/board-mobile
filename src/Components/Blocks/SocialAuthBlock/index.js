import { compose } from 'redux'
import { connect } from 'react-redux'

import withNamespaces from 'Components/HOC/withNamespaces'

import { federatedSignIn } from 'Redux/actions/auth'

import Component from './SocialAuthBlock'

const actions = {
  onFederatedSignIn: federatedSignIn,
}

export default compose(
  withNamespaces,
  connect(
    null,
    actions,
  ),
)(Component)
