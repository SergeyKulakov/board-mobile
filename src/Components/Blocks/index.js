import SliderAdvertising from './SliderAdvertising'
import LanguageModal from './LanguageModal'
import ServicesList from './ServicesList'
import ProviderProfileCard from './ProviderProfileCard'
import ServicesAutocomplete from './ServicesAutocomplete'
import JobsList from './JobsList'
import ServiceProvidersList from './ServiceProvidersList'
import SocialAuthBlock from './SocialAuthBlock'
import RequestsJobsList from './RequestsJobsList'
import SelectDateModal from './SelectDateModal'
import ChangePasswordModal from './ChangePasswordModal'
import SignInForm from './SingInForm'
import JobCard from './JobCard'
import DoerInfo from './DoerInfo'
import NavigationMenu from './NavigationMenu'
import NotificationCard from './NotificationCard'
import ReviewProfile from './ReviewProfile'
import ReviewCard from './ReviewCard'
import ButtonsGroup from './ButtonsGroup'
import ApplyServiceProviderCard from './ApplyServiceProviderCard'
import LinkPreview from './LinkPreview'
import LicenseAgreement from './LicenseAgreement'
import BannerAdvertising from './BannerAdvertising'
import MapWrapper from './MapWrapper'
import TrackMap from './TrackMap'

export * from './InputModal'
export * from './ScreenLoader'
export * from './ServerError'
export * from './SocialButton'
export * from './LanguageButton'
export * from './VideoPlayer'
export * from './JobImages'
export * from './PeopleList'
export * from './DatePicker'
export * from './UploadImages'
export * from './ApplyJobModal'
export * from './MapSlider'
export * from './ChatCard'
export * from './ChatMessage'
export * from './SettingRow'
export * from './RequestsSPList'
export * from './Map'

export {
  ServicesList,
  LanguageModal,
  SliderAdvertising,
  ProviderProfileCard,
  ServicesAutocomplete,
  JobsList,
  ServiceProvidersList,
  SocialAuthBlock,
  RequestsJobsList,
  SelectDateModal,
  ChangePasswordModal,
  SignInForm,
  JobCard,
  DoerInfo,
  NavigationMenu,
  NotificationCard,
  ReviewProfile,
  ReviewCard,
  ButtonsGroup,
  ApplyServiceProviderCard,
  LinkPreview,
  LicenseAgreement,
  BannerAdvertising,
  MapWrapper,
  TrackMap,
}
