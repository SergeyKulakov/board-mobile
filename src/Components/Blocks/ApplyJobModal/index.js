import React from 'react'
import PropTypes from 'prop-types'
import { Modal, Button } from 'Components/UI'

import { ActivityIndicator } from 'react-native'

import { getText } from './config'
import {
  Container,
  Title,
  ButtonBlock,
  LoaderContainer,
  TitleWrapper,
  styles,
} from './style'

const ApplyJobModal = ({
  visible,
  onCancel,
  onSubmit,
  loading,
  text: propText,
}) => {
  const text = propText || getText()

  return (
    <Modal onCancelPress={onCancel} visible={visible}>
      <Container>
        <TitleWrapper>
          <Title>{text.title}</Title>
        </TitleWrapper>
        {loading ? (
          <LoaderContainer>
            <ActivityIndicator {...styles.Loader} />
          </LoaderContainer>
        ) : (
          <ButtonBlock>
            <Button {...styles.SkipButton} onClick={onCancel}>
              {text.skip}
            </Button>
            <Button {...styles.ApplyButton} onClick={onSubmit}>
              {text.apply}
            </Button>
          </ButtonBlock>
        )}
      </Container>
    </Modal>
  )
}

ApplyJobModal.propTypes = {
  text: PropTypes.shape({
    title: PropTypes.string,
    subTitle: PropTypes.string,
    apply: PropTypes.string,
    skip: PropTypes.string,
  }),
  visible: PropTypes.bool,
  loading: PropTypes.bool,
  onSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
}

export { ApplyJobModal }
