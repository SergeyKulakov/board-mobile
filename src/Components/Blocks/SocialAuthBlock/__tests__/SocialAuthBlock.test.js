import React from 'react'
import { shallow } from 'enzyme'
import chai from 'chai'

import 'jest-styled-components'

import { SocialButton } from 'Components/Blocks'
import Component from '../SocialAuthBlock'

describe('SocialAuthBlock', () => {
  const defaultProps = {
    toast: () => null,
    onFederatedSignIn: () => null,
    getError: () => null,
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('call federation sign in', () => {
    it('should call onFederatedSignIn', () => {
      const props = {
        ...defaultProps,
        onFederatedSignIn: jest.fn(),
        toast: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      expect(props.onFederatedSignIn).not.toHaveBeenCalled()
      expect(props.toast).not.toHaveBeenCalled()

      const response = { url: 'url' }

      wrapper.instance().handleUrl(response)

      expect(props.onFederatedSignIn).toHaveBeenCalledTimes(1)
    })
    it('successful federation sign in', done => {
      const props = {
        ...defaultProps,
        onFederatedSignIn: jest.fn(({ callback }) => {
          setTimeout(() => {
            callback({})
          }, 1000)
        }),
        toast: jest.fn(),
        getError: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      const response = { url: 'url' }

      wrapper.instance().handleUrl(response)

      expect(props.onFederatedSignIn).toHaveBeenCalledTimes(1)

      setTimeout(() => {
        expect(props.toast).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()
        expect(props.onFederatedSignIn).toHaveBeenCalledTimes(1)
        done()
      }, 1100)
    })

    it('failure federation sign in', done => {
      const data = {
        error: {
          payload: {
            code: 'some code',
          },
        },
      }
      const props = {
        ...defaultProps,
        onFederatedSignIn: jest.fn(({ callback }) => {
          setTimeout(() => {
            callback(data)
          }, 1000)
        }),
        toast: jest.fn(),
        getError: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      const response = { url: 'url' }

      wrapper.instance().handleUrl(response)

      expect(props.onFederatedSignIn).toHaveBeenCalledTimes(1)
      expect(props.toast).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()

      setTimeout(() => {
        expect(props.toast).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledWith(data.error)
        expect(props.onFederatedSignIn).toHaveBeenCalledTimes(1)
        done()
      }, 1100)
    })
  })

  it('SocialButton press', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    wrapper.instance().handleLinkClick = jest.fn()

    chai.expect(wrapper.find(SocialButton)).to.lengthOf(3)

    expect(wrapper.instance().handleLinkClick).not.toHaveBeenCalled()

    wrapper.find(SocialButton).forEach(el => {
      el.props().onClick()
    })

    expect(wrapper.instance().handleLinkClick).toHaveBeenCalledTimes(3)
  })
})
