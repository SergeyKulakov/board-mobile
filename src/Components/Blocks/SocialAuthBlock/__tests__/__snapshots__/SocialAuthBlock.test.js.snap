// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialAuthBlock render snapshot 1`] = `
<Styled(View)>
  <SocialButton
    color="#39539d"
    icon="facebook-f"
    onClick={[Function]}
  />
  <SocialButton
    color="#ed381d"
    icon="google"
    onClick={[Function]}
  />
  <SocialButton
    color="#0074bb"
    icon="linkedin"
    onClick={[Function]}
  />
</Styled(View)>
`;
